sdk-java/src/test/java/com/spotify/confidence/ConfidenceResourceManagementTest.java

```diff
@@ -21,44 +20,18 @@ public void setup() {
     root = Confidence.create(fakeEngine, fakeFlagResolverClient);
   }
 
-  @Test
-  public void testCloseChildShouldThrowFromResolveFlags() throws IOException {
-    final Confidence child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
-    child.close();
-    assertThrows(IllegalStateException.class, () -> child.resolveFlags("test", false).get());
-  }
-
   @Test
   public void testCloseChildShouldNotCloseParentEngine() throws IOException {
-    final Confidence child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
+    final EventSender child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
     child.close();
     assertFalse(fakeEngine.closed);
     assertFalse(fakeFlagResolverClient.closed);
   }
 
   @Test
   public void testCloseChildShouldNotThrowFromSend() throws IOException {
-    final Confidence child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
+    final EventSender child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
     child.close();
     child.track("Test");
   }
-
-  @Test
-  public void testCloseChildShouldNotAffectParent()
-      throws IOException, ExecutionException, InterruptedException {
-    final Confidence child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
-    child.close();
-    root.resolveFlags("test", false).get();
-    root.track("test", ConfidenceValue.of(Map.of("messageKey", ConfidenceValue.of("parent"))));
-  }
-
-  @Test
-  public void testCloseParentShouldAffectChild() throws IOException {
-    final Confidence child = root.withContext(Map.of("child-key", ConfidenceValue.of("child")));
-    root.close();
-    assertThrows(IllegalStateException.class, () -> child.resolveFlags("test", false).get());
-    assertThrows(IllegalStateException.class, () -> root.resolveFlags("test", false).get());
-    assertTrue(fakeEngine.closed);
-    assertTrue(fakeFlagResolverClient.closed);
-  }
```

Comment:
The child of `root.withContext` is only an EventSender, doesn't allow resolves. I think this is a limitation we actually wanted to introduce?